### PR TITLE
Organization/31/Format and Restructure code

### PR DESCRIPTION
**Pull request by brendanehrman1**
- Add pre-commit hooks to automatically format code to compliance with pep8
    - Reformat all files using hooks for consistency
- Simplify data processing logic using Pandas
   - sort_companies now uses DataFrame.sort_values() and writes CSV with quoting=QUOTING_ALL
   - import_companies now uses DataFrame.merge() to easily add scrape configurations without overwrite
   - scraper now writes to CSV for consistency and to greatly improve future data processing/filtering
- Restructure codebase to reduce clutter
   - Move frontend, backend, data into new pipelines folder
   - Move models into single file (more future models will also be placed)
   - Move cli handling into main

### CODE REVIEW

File: agtern/__main__.py [hunk 1]
```diff
@@ -1,20 +1,40 @@
...
-    except KeyboardInterrupt: # Ctrl+C
-        pass # Do nothing and hide Traceback
+    except KeyboardInterrupt:  # Ctrl+C
+        pass  # Do nothing and hide Traceback
+
+
+def run_cli(args=None):
+    parser = ArgumentParser(prog="AgTern")
+    parser.add_argument("--update-companies", action="store_true")
+    parser.add_argument("--show-scraper", action="store_true")
```
Comment (johndpjr) [inline agtern/__main__.py:26]:
```suggestion
    parser.add_argument("--headless", action="store_true")
```

File: agtern/__main__.py [hunk 2]
```diff
@@ -1,20 +1,40 @@
...
+    parser.add_argument("--noscrape", action="store_true")
+    args = parser.parse_args(args)
+
+    if args.update_companies:
+        print("INFO: Updating company info...")
+        sort_companies()
+        import_companies()
+    else:
+        print("INFO: Starting program...")
+        main(noscrape=args.noscrape, headless_scraper=not args.show_scraper)
```
Comment (johndpjr) [inline agtern/__main__.py:36]:
```suggestion
        main(noscrape=args.noscrape, headless_scraper=args.headless)
```

File: agtern/__main__.py [hunk 3]
```diff
@@ -1,20 +1,40 @@
...
+    except KeyboardInterrupt:  # Ctrl+C
+        pass  # Do nothing and hide Traceback
+
+
+def run_cli(args=None):
+    parser = ArgumentParser(prog="AgTern")
+    parser.add_argument("--update-companies", action="store_true")
+    parser.add_argument("--show-scraper", action="store_true")
+    parser.add_argument("--noscrape", action="store_true")
+    args = parser.parse_args(args)
```
Comment (johndpjr) [inline agtern/__main__.py:28]:
```suggestion
    args = parser.parse_args()
```

File: agtern/__main__.py [hunk 4]
```diff
@@ -1,20 +1,40 @@
...
     app = Application()
     try:
         app.mainloop()
-    except KeyboardInterrupt: # Ctrl+C
-        pass # Do nothing and hide Traceback
+    except KeyboardInterrupt:  # Ctrl+C
+        pass  # Do nothing and hide Traceback
+
+
+def run_cli(args=None):
```
Comment (johndpjr) [inline agtern/__main__.py:23]:
```suggestion
def run_cli():
```

File: agtern/__main__.py [hunk 5]
```diff
@@ -1,20 +1,40 @@
-from agtern.gui import Application
-from agtern.backend import start_server
+from argparse import ArgumentParser
+
+from .gui import Application
+from .pipelines import import_companies, sort_companies, start_server
 
 HEADLESS_SCRAPER = True
```
Comment (johndpjr) [inline agtern/__main__.py:6]:
```suggestion
```

File: agtern/__main__.py [hunk 6]
```diff
@@ -1,20 +1,40 @@
...
 
-def main( noscrape: bool = True, headless_scraper = None ):
+
+def main(noscrape: bool = True, headless_scraper=None):
```
Comment (johndpjr) [inline agtern/__main__.py:9]:
```suggestion
def main(noscrape=True, headless_scraper=True):
```

File: agtern/__main__.py [hunk 7]
```diff
@@ -1,20 +1,40 @@
...
     if not noscrape:
         if headless_scraper is None:
```
Comment (johndpjr) [inline agtern/__main__.py:11]:
```suggestion
```

File: agtern/__main__.py [hunk 8]
```diff
@@ -1,20 +1,40 @@
...
-            start_server( HEADLESS_SCRAPER )
+            start_server(HEADLESS_SCRAPER)
         else:
```
Comment (johndpjr) [inline agtern/__main__.py:13]:
```suggestion
```

**Review by johndpjr — COMMENTED** [review]
Good job Brendan! This gets us far in standardizing our codebase. A few things:
* I think all of our models should be put into an `agtern/models` folder just like before. Adding another path to `agtern/utils/models` adds more complexity than is needed. Also, utils and models are two separate concepts, so I would just have one `utils` and one `models` folder that are separated.
* Should `import_companies.py` and `sort_companies.py` be inside the `utils` folder since they are utility scripts that act on the data (instead of being inside `data` in `utils`)?

**Comment by brendanehrman1** [comment]
@johndpjr Makes sense! Let me know (along with @JeremyEastham) if there are any more changes that I should address in this PR. As a side note, it would be good to merge this PR before working on any future changes.

**Review by johndpjr — APPROVED** [review]
LGTM!

**Pull request merged**